app/controllers/reservations_controller.rb

```diff
@@ -1,3 +1,23 @@
-def my_reservations
-  @reservations = current_user.reservations
+# app/controllers/reservations_controller.rb
+class ReservationsController < ApplicationController
+  # before_action :authenticate_user!
+
+  def my_reservations
+    @reservations = User.find(session[:user_id]).reservations
+  end
+
+  def create
+    @reservation = User.find(session[:user_id]).reservations.create(reservation_params)
+    if @reservation.save
+      redirect_to my_reservations_path, notice: 'Reservation was successfully created.'
+    else
```

Comment:
As you create **resources :reservations**, the reservation path should be reservations_path